test/libsolidity/syntaxTests/inlineAssembly/transient_storage_invalid_pre_cancun.sol

```diff
@@ -0,0 +1,13 @@
+contract C {
+    function f() public {
+        assembly {
+            tstore(0, 13)
+            tload(0)
+        }
```

Comment:
I had a check like that, but as you said, it does not work, so I just removed it. I don't quite remember the reason, but will try to find out and make it behave like `basefee`.

Reply:
@nikola-matic investigated that and it apparently it may be caused by `cancun` not being the current version. I'm going to take a closer look later since I'll encounter this in my `mcopy` PR anyway, but for now you can leave a is after all.

Reply:
Actually, I'd still add the check in `AsmAnalysis.cpp` even if we cannot test it properly yet. The current test won't fail when we make it the default and we'll very likely forget then.

Reply:
I added the check and whitelisted it in `error_codes.py`.

Reply:
Isn't that check contradictory with allowing `tstore` and `tload` as identifiers before cancun? I remembered that was also a reason why I removed the check after implementing it the first time...

Reply:
Looks like it isn't. I think that this check is meant only to provide a more detailed error in cases where the compilation would fail anyway due to it being an undeclared identifier. `validateInstructions()` [used to be called `warnOnInstructions()`](https://github.com/ethereum/solidity/commit/2e6cbd87182b7960db1cb36158b6a7978f9c2649) and it probably intentionally [uses the default EVM version](https://github.com/ethereum/solidity/blob/v0.8.23/libyul/AsmAnalysis.cpp#L673) (rather than the selected one) to look up the opcode. This way the error is only issued when the undeclared identifier is a builtin on the default EVM but not on the currently selected one.

And it only catches undeclared identifiers because the validation is performed only in two cases: [for built-ins present in the selected EVM version](https://github.com/ethereum/solidity/blob/v0.8.23/libyul/AsmAnalysis.cpp#L327) (but these will always pass without error [even if not present in the default EVM](https://github.com/ethereum/solidity/blob/v0.8.23/libyul/AsmAnalysis.cpp#L677)) and [for undeclared identifiers](https://github.com/ethereum/solidity/blob/v0.8.23/libyul/AsmAnalysis.cpp#L356). A [properly declared identifier](https://github.com/ethereum/solidity/blob/v0.8.23/libyul/AsmAnalysis.cpp#L330-L353) (shadowing the built-in) is the third case that does not go through this check.

Still, this is not all that obvious unless you read the code carefully. I think we could use some more comments there explaining this. Or at least better naming.

Reply:
By the way, I just realized that instead of adding this error now (when we cannot really test it and have to ignore it via `error-codes.py`) you could also just make it an assert. Then it will fail when we make `cancun` the default and we'll add the error then. That might be better than having an unreachable error.

Reply:
Right. I made it an assertion. See last commit ;)

Reply:
I changed the error to be an assert.